README.md

```diff
@@ -611,3 +611,59 @@ status:
   binding:
     name: kafka-eftg9
 ```
+
+## Multi-Application Bindings
+
+The `ServiceBinding` resource is limited by the [Service Binding](#service-binding) spec to bind a single service by-name to a single application by-name. If the application's name is generated and not known in advance, the binding implicitly must be created after the application. While level-based reconciliation will converge on the desired state, the application will roll out initially without the bound service. Depending on the application this may cause a crash loop or a misconfiguration (e.g. using an in-memory database designed for development in a production environment). By switching the application reference from using a name to a label selector the service can be bound to the application resource as it is created.
+
+A `ServiceBinding` **MAY** define the application reference by-name or by-[label selector][ls]. A name and selector **MUST NOT** be defined in the same reference.
+
+As label selectors are inherently queries that return zero-to-many resources, it is **RECOMMENDED** that `ServiceBinding` authors use a combination of labels that yield a single resource, but implementors of this extension **SHOULD** handle each matching resource as if it was specified by name in a distinct `ServiceBinding` resource. Partial failures **SHOULD** be aggregated and reported on the binding status's `Ready` condition.
```

Comment:
```suggestion
As label selectors are inherently queries that return zero-to-many resources, it is **RECOMMENDED** that `ServiceBinding` authors use a combination of labels that yield a single resource, but implementors of this extension **MUST** handle each matching resource as if it was specified by name in a distinct `ServiceBinding` resource. Partial failures **MUST** be aggregated and reported on the binding status's `Ready` condition.
```

Reopening https://github.com/application-stacks/service-binding-specification/pull/55#discussion_r442924014 based on discussion on the interlock call